packages/loot-core/src/server/accounts/rules.ts

```diff
@@ -19,11 +19,11 @@ import {
   ungroupTransaction,
 } from '../../shared/transactions';
 import { fastSetMerge } from '../../shared/util';
-import { RuleConditionEntity } from '../../types/models';
+import { RuleConditionEntity, RuleEntity } from '../../types/models';
 import { RuleError } from '../errors';
 import { Schedule as RSchedule } from '../util/rschedule';
 
-function assert(test, type, msg) {
+function assert(test: unknown, type: string, msg: string): asserts test {
```

Comment:
Wow I didn't know about `asserts`, this is great!

Reply:
Wondering your thoughts on doing `test: boolean` and explicitly converting at call sites? My thinking:
- Pro: this function is better documented, as `test` is always coerced into a boolean
- Con: have to modify all call sites and do minor unnecessary conversion

Reply:
I would think it's unnecessary. `assert(!!value, ...)` is not much clearer than `assert(value, ...`.

Reply:
fair enough!